int32 flag

---
float64[] trans
float64[] quat